topdown/net.go

```diff
@@ -0,0 +1,44 @@
+package topdown
+
+import (
+	"net"
+
+	"github.com/open-policy-agent/opa/ast"
+	"github.com/open-policy-agent/opa/topdown/builtins"
+)
+
+// resolv is used for name resolution in builtinLookupIPAddr.
+// We'using PreferGo: true to get consistent behaviour across different platforms.
+var resolv = net.Resolver{PreferGo: true}
+
+type lookupIPAddrCacheKey string
+
+func builtinLookupIPAddr(bctx BuiltinContext, operands []*ast.Term, iter func(*ast.Term) error) error {
+	name, err := builtins.StringOperand(operands[0].Value, 1)
+	if err != nil {
+		return err
+	}
+
+	key := lookupIPAddrCacheKey(name)
+	if val, ok := bctx.Cache.Get(key); ok {
+		return iter(val.(*ast.Term))
+	}
+
+	addrs, err := resolv.LookupIPAddr(bctx.Context, string(name))
+	if err != nil { // handle networking errors differently? Halt?
```

Comment:
Hrrm it looks like the errors look pretty much the same, for

- network not available:

  `lookup localhoxt on 192.168.65.5:53: dial udp 192.168.65.5:53: connect: network is unreachable (*net.DNSError)`
- host not found:

  `lookup xemaple.com on 192.168.178.1:53: no such host (*net.DNSError)`
- an invalid ip addr passed:
 
  `lookup 10.0.0.500: no such host (*net.DNSError)`